CompactPhp/NonstandardReal.lean

```diff
@@ -0,0 +1,289 @@
+import Mathlib.ModelTheory.Satisfiability
+import Mathlib.Data.Fintype.Card
+import Mathlib.Data.Real.Basic
+open FirstOrder
+open FirstOrder.Language
+open FirstOrder.Language.Theory
+open Real
+
+section NonStandard
+
+-- The reals extended with a non-standard element
+/-
+Not that crazy. We are used to having such a type in interval arithmetic
+Floats have +inf and -inf representable.
+We could define normal addition,subtraction functions
+over this stuff
+
+for exmaple
+def add x y :=
```

Comment:
This is incorrect: a non standard integer/real cannot be equal to itself when added to a non-zero number.

Reply:
Ah interesting.
I guess I think of the nonstandard reals interesting as a way to extend more familiar number systems with infinities in a cogent way.
- Ordinals fit this. Addition becomes non commutative on infinite objects
- The truncated float style inf fit this. truncated addition makes this easier to represent.
- the nonstandard/hyperintegers are another thing. I suppose the infinite elements need to obey the same algebraic laws and the ordinary number.
- and the surreals are yet another thing.